### ISSUE TITLE
Check fails with "base resource type not found" when the resource name is the same as the resource type

### ISSUE THREAD

**Issue by thomasd-** [concourse/concourse#4663]
## Bug Report

When a resource has the same name as its resource type, every check fails with the error "base resource type not found", whether the check is triggered via the cli, via the web ui or via a webhook.

## Steps to Reproduce

[Test pipeline](https://gist.github.com/thomasd-/b1e1cd0ab96e4af664eca51ae60fcf60)

```
fly -t target sp -p test -c pipeline.yml
fly -t target up -p test
fly -t target cr -r test/cron-resource
```

## Expected Results

The check should happen and return no error.

## Actual Results

```
error: base resource type not found: cron-resource
```

## Version Info

* Concourse version: 5.6.0
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: Azure
* Browser (if applicable): Not relevant
* Did this used to work? Yes, at least up ton Concourse v 5.3.0

**Comment by elsmorian** [comment]
We have certainly seen this too via Github web hooks on a PullRequest resource, renaming the resource to something different than the resource type fixed it. Pretty sure this broke when we moved to v5.6.0

cc @pikesley

**Comment by pivotal-jamie-klassen** [comment]
I believe this is a duplicate of #4599, and has been fixed on master in #4605 - should be in the next minor.

**Comment by myles-mcdonnell** [comment]
I set a pull-request resource to the same name and triggered this problem on 6.3.  I have since renamed the resource, restarted the box (I'm running a binary install on EC2 with RDS) and upgraded to 6.7 and still no dice.  About to blow away my DB.  I get the feeling something is hanging around from the initial plan conf causing this.  I have other plans on the box that work with PR webhooks fine.
